docs/en/middleware/supplemental.md

```diff
@@ -0,0 +1,521 @@
+
+### `getRegistry`
+
+Provides access to the widget's own `Registry` instance, as well as the root application `Registry` if required, via a handler interface.
+
+**Note:** The registry is an advanced concept not typically required when writing Dojo applications. It is mainly used internally by the framework to implement more advanced user-facing functionality such as [Dojo stores](../stores/basic-usage.md).
+
+**API:**
+
+```ts
+import getRegistry from '@dojo/framework/core/vdom';
+```
+
+-   `getRegistry(): RegistryHandler | null`
+    -   Returns the `RegistryHandler` for the current widget, or `null` if the widget has not yet been fully initialized.
+
+## Optional middleware
+
+Dojo provides a variety of optional middleware that widgets can include when needing to implement specific requirements.
+
+### `block`
+
+Allows widgets to execute modules known as **blocks** within Node.js at build time. Typically used as part of build-time rendering.
+
+Described in detail in the [Building](../building/supplemental.md#build-time-rendering) reference guide.
+
+**API:**
+
+```ts
+import block from '@dojo/framework/core/middleware/block';
+```
+
+-   `block<T extends (...args: any[]) => any>(module: T)`
+    -   Executes the specified block module and returns its result.
+
+### `breakpoint`
+
+Allows widgets to determine a specific width breakpoint that is matched given the current width of one of their virtual nodes. Useful when creating widgets that can adapt to a variety of display widths, such as widgets that work at both mobile and desktop resolutions.
+
+Composes the [`resize`](#resize) middleware to obtain the element width and to automatically invalidate the widget when its width is adjusted.
+
+**Note:** If no custom width breakpoints are given, Dojo will default to the following set:
+
+-   `SM`: 0
+-   `MD`: 576
+-   `LG`: 768
+-   `XL`: 960
+
+**API:**
+
+```ts
+import breakpoint from '@dojo/framework/core/middleware/breakpoint';
+```
+
+```ts
+interface Breakpoints {
+	[index: string]: number;
+}
+```
+
+-   `breakpoint.get(key: string | number, breakpoints: Breakpoints = defaultBreakpoints)`
+    -   Returns the breakpoint that the widget's specified output node (identified by its `key`) matches, given the node's current width. Custom breakpoints can be provided through the `breakpoints` argument. The return value is an object containing a `breakpoint` property, identifying the name of the breakpoint that was matched, and a `contentRect` property which contains the same value as calling `resize.get(key)` would return.
+
+When using the same set of breakpoints in many locations, it is easier to define the set once rather than have to pass it in to every `breakpoint.get()` call. Applications can define their own custom breakpoint middleware with appropriate defaults via:
```

Comment:
```suggestion
When using the same set of breakpoints in many locations, it is easier to define the set once rather than needing to pass it to every `breakpoint.get()` call. Applications can define their own custom breakpoint middleware with appropriate defaults via:
```